packages/react-native-windows-init/src/Cli.ts

```diff
@@ -334,32 +327,127 @@ function installReactNativeWindows(
   );
 }
 
-function getRNWInitVersion(): string {
+/**
+ * Sanitizes the given option for telemetery.
+ * @param key The key of the option.
+ * @param value The unsanitized value of the option.
+ * @returns The sanitized value of the option.
+ */
+function optionSanitizer(key: string, value: any): any {
+  switch (key) {
+    case 'namespace':
+      return value === undefined ? false : true;
+    default:
+      return value === undefined ? false : value;
+  }
+}
+
+/**
+ * Sets up and starts the telemetry gathering for the CLI command.
+ */
+async function startTelemetrySession() {
+  if (!argv.telemetry) {
+    if (argv.verbose) {
+      console.log('Telemetry is disabled');
+    }
+    return;
+  }
+
+  await Telemetry.setup();
+
+  const sanitizedOptions = yargsOptionsToOptions(argv, optionSanitizer);
+  const sanitizedDefaultOptions = yargsOptionsToOptions(
+    yargsParser.parse(''),
+    optionSanitizer,
+  );
+  const sanitizedArgs = optionsToArgs(sanitizedOptions, process.argv);
+
+  const startInfo: CommandStartInfo = {
+    commandName: 'react-native-windows-init',
+    args: sanitizedArgs,
+    options: sanitizedOptions,
+    defaultOptions: sanitizedDefaultOptions,
+  };
+
+  Telemetry.startCommand(startInfo);
+}
+
+/**
+ * Adds the new project's telemetry info by calling and processing `react-native config`.
+ */
+async function addProjectInfoToTelemetry() {
+  if (!Telemetry.isEnabled()) {
+    return;
+  }
+
   try {
-    const pkgJson = require('../package.json');
-    if (
-      pkgJson.name === 'react-native-windows-init' &&
-      pkgJson.version !== undefined
-    ) {
-      return pkgJson.version;
+    const config = JSON.parse(
+      execSync('npx react-native config', {
+        stdio: ['ignore', 'pipe', 'ignore'],
+      }).toString(),
+    );
```

Comment:
consider not spawning a new nodejs instance and instead require config

Reply:
We need the whole infra of the `react-native config` command to run in order to get data like what other platforms are supported.